sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpSymbolUsageCollector.cs

```diff
@@ -179,10 +179,24 @@ private SymbolAccess ParentAccessType(SyntaxNode node)
 
         public override void VisitObjectCreationExpression(ObjectCreationExpressionSyntax node)
         {
-            UsedSymbols.UnionWith(GetSymbols(node, x => true));
+            if (!IsInInitializerExpression(node))
+            {
+                UsedSymbols.UnionWith(GetSymbols(node, x => true));
+            }
+
             base.VisitObjectCreationExpression(node);
         }
 
+        private static bool IsInInitializerExpression(SyntaxNode node)
+        {
+            var parentKind = node.Parent.Kind();
+
+            return parentKind == SyntaxKind.ArrayInitializerExpression
```

Comment:
You've added the perf test, and I would add some more UTs to test the functionality, that the unused private fields are detected and issues are raised for the syntaxes covered here, when the private field exists and it's not used.


Including edge cases like
```
public class Foo
{
	private string x;
	private string[] y;
	public Foo()
	{
		y = new string[] { x = new String("foo".ToCharArray()) };
		new Qix(x = new String("bar".ToCharArray()));
		new Qix(x : x = new String("X".ToCharArray()));
		new List<string> { x = new String("Y".ToCharArray()) };
	}
}

public class Qix
{
	private string y;
	public Qix(string x)
	{
		y = x;
	}
}

public class Matrix
{
    private double[,] storage = new double[3, 3];

    public double this[int row, int column]
    {
        // The embedded array will throw out of range exceptions as appropriate.
        get { return storage[row, column]; }
        set { storage[row, column] = value; }
    }
}
```
unless we already test them

and also with [object initializers](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/object-and-collection-initializers#object-initializers) w/ and w/o anonymous types

and also check what happens when creating an object in a LINQ query - as a common scenario

Reply:
This is no longer valid since I've changed the approach.

Reply:
BTW, some of these tests would still be valuable to have , in case we don't cover these sort of syntaxes

Reply:
Since this is not directly related to this PR I would like to postpone this to keep this PR focused on the problem at hand. It is already big enough.